src/modules/Selectors.jl

```diff
@@ -0,0 +1,151 @@
+"""
+An extensible code selection interface.
+
```

Comment:
I would have a quick overview of the interface here, I was initially somewhat confused with the example.

Perhaps something along the lines of:

> It makes writing code that has to dispatch based on a condition easy.
> The call to `selector(::T, x)` chooses the appropriate `runner` to call based on which of the `matcher`s
> returns `true`.

Perhaps it would be useful to say that `selector` effectively is a nice automatically generated if statement?
```julia
function selector(x)
    if matcher(::First, x)
        runner(::First, x)
    elseif matcher(::Second, x)
        runner(::Second, x)
    ...
```